lib/ecto/repo/queryable.ex

```diff
@@ -428,6 +459,26 @@ defmodule Ecto.Repo.Queryable do
     Query.where(queryable, [], ^Enum.to_list(clauses))
   end
 
+  defp query_for_reload([head| _] = structs, opts) do
+    assert_structs!(structs)
+
+    schema = head.__struct__
+
+    case schema.__schema__(:primary_key) do
+      [pk] ->
+        keys = Enum.map(structs, &get_pk!(&1, pk))
+        preloads = Keyword.get(opts, :preload, [])
```

Comment:
Honestly, I don't think we need to support this option. Doing a `|> Repo.preload(...)` after is only very few characters more verbose than `, preload: `.

Reply:
@josevalim wouldn't `|> Repo.preload()` imply another trip to the database while `from(..., preload: ...)` does it in one pass? Or they're always equivalent?

Reply:
Preload always issues separates queries, that’s what it is about, so both forms are equivalent!

Reply:
That's nice to know! I thought the form in `Ecto.Query` would optimize into a join of some kind